### PR TITLE
Chaos Engineering: Create a simulated unreliable K8s client

### PR DESCRIPTION
**Pull request by Rizwana777**
#### Description:
- Create a simulated unreliable K8s client

#### Link to JIRA Story (if applicable):
https://issues.redhat.com/browse/GITOPSRVCE-284

### CODE REVIEW

File: backend-shared/util/chaosclient.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+func shouldSimulateFailure() error {
+
+	if !isEnvExist("UNRELIABLE_CLIENT_FAILURE_RATE") {
+		return nil
+	}
+
+	unreliableClientFailureRate := os.Getenv("UNRELIABLE_CLIENT_FAILURE_RATE")
+	unreliableClientFailureRateValue, err := strconv.Atoi(unreliableClientFailureRate)
```
Comment (jgwest) [inline backend-shared/util/chaosclient.go:184]: @Rizwana777 , following up from standup, try this:
```go
	unreliableClientFailureRate := os.Getenv("UNRELIABLE_CLIENT_FAILURE_RATE")
	unreliableClientFailureRateValue, err := strconv.Atoi(unreliableClientFailureRate)
	if err != nil {
		return err
	}
	
	// Convert it to a decimal point, e.g. 50% to 0.5 
	unreliableClientFailureRateValue = unreliableClientFailureRateValue / 100
	
	// return an error randomly x% of the time
	if rand.Float64() < unreliableClientFailureRateValue {
		return fmt.Errorf("simulated K8s error")
	}
	
	return nil
```

Comment (Rizwana777) [reply backend-shared/util/chaosclient.go:184]: Thanks @jgwest

**Review by jgwest — APPROVED** [review]
Works great, I'm already using it to check our code/tests. Thanks @Rizwana777!

**Pull request merged**